Remove bench dependency to fix setup re-run issue (#95)

ANAMASGARD opened the pull request:
Test demonstrates Issue #95: bench::mark does not re-run setup between iterations, causing incorrect timings for reference-modifying functions like data.table::setkey().

Shows 9.8x discrepancy between first iteration (actual sort) and subsequent iterations (just verification). Documents current behavior.

<img width="3072" height="1840" alt="Screenshot From 2025-10-28 20-44-45" src="https://github.com/user-attachments/assets/b4ac8c1e-80e5-4fce-b290-2fe360debea3" />

ANAMASGARD committed:
Add test documenting bench::mark limitation

Test demonstrates Issue #95: bench::mark does not re-run setup
between iterations, causing incorrect timings for reference-modifying
functions like data.table::setkey().

Shows 9.8x discrepancy between first iteration (actual sort) and
subsequent iterations (just verification). Documents current behavior.

tdhock commented on tests/testthat/test-reference-modification.R:
```diff
@@ -0,0 +1,31 @@
+  cat(sprintf("Discrepancy: First iteration is %.1fx slower than rest\n", ratio))
+  expect_gt(ratio, 3, label = sprintf("Issue #95 confirmed: bench::mark does not re-run setup (ratio=%.1f)", ratio))
+  cv <- sd(times_each) / mean(times_each)
+  expect_lt(cv, 0.5, label = sprintf("When setup runs each time, timings are consistent (CV=%.3f)", cv))
```
please remove these tests, which are not robust because of the dependence on magic numbers (arbitrary constants 3 and 0.5).
it would be better to test based on a counter, something like this
```r
> count_list <- list(setup=0, expr=0)
> atime::atime(setup=count_list$setup <<- count_list$setup+1, myexpr=count_list$expr <<- count_list$expr+1)
atime list with 20 measurements for
myexpr(N=2 to 1048576)
> count_list
$setup
[1] 20

$expr
[1] 220

```

ANAMASGARD committed:
Replace timing-based test with counter-based test for Issue #95

Uses counter to demonstrate bench::mark limitation: setup runs once per N
value while expr runs multiple times per N. This is more robust than
timing-based tests that depend on arbitrary thresholds.

Test shows setup=5 vs expr=55, confirming setup doesn't re-run between
bench::mark iterations.
